src/ucp/api/ucp.h

```diff
@@ -1747,6 +1845,74 @@ ucs_status_ptr_t ucp_tag_send_sync_nb(ucp_ep_h ep, const void *buffer, size_t co
 
 /**
  * @ingroup UCP_COMM
+ * @brief Non-blocking stream receive operation.
+ *
+ * This routine receives a data that is described by the local address @a
+ * buffer, size @a count, and @a datatype object on the endpoint @a ep. The
+ * routine is a non-blocking and therefore returns immediately. The receive
+ * operation is considered completed when the message is delivered to the @a
+ * buffer. In order to notify the application about completion of the receive
+ * operation the UCP library will invoke the call-back @a cb when the received
+ * data is in the receive buffer and ready for application access. If the
+ * receive operation cannot be stated the routine returns an error.
```

Comment:
Stream operations are strongly p2p. Ordering should be guarantee in the stream:

- if the are posted `ucp_stream_recv_nb` ops, firstly these buffers will be used for data placement in the posting order and callbacks should be called in the same order.
- otherwise `ucp_stream_worker_poll` will return the EP there are available data to receive and this is user's choice how to obtain it.
- in case of mixed usage of `ucp_stream_recv_nb` and `ucp_stream_recv_data_nb`: firstly, posted requests will be consumed then user can obtain the data by `ucp_stream_recv_data_nb` directly or post new requests by `ucp_stream_recv_nb`